src/borg/testsuite/crypto.py

```diff
@@ -248,3 +258,120 @@ def test_hkdf_hmac_sha512_5(self):
 
         okm = hkdf_hmac_sha512(ikm, salt, info, l)
         assert okm == bytes.fromhex('1407d46013d98bc6decefcfee55f0f90b0c7f63d68eb1a80eaf07e953cfc0a3a5240a155d6e4daa965bb')
+
+
+def test_decrypt_key_file_argon2_aes256_ctr_hmac_sha256(monkeypatch):
+    monkeypatch.delenv('BORG_TESTONLY_MOCK_KDF')
+    plain = b'hello'
+    # echo -n "hello, pass phrase" | argon2 saltsaltsaltsalt -id -t 3 -m 16 -p 4 -l 64 -r
+    enc_key = bytes.fromhex('3dd855b778ba292eda7bf708a9ea111ee99c5c45d2e9a2773d126de46d344410')
+    mac_key = bytes.fromhex('0b0b65fdccaea7cf5b9a6214cd867983e2326abeccedf1dceb1feee0ae74075b')
+    ae_cipher = AES256_CTR_HMAC_SHA256(
+        iv=0, header_len=0, aad_offset=0,
+        enc_key=enc_key,
+        mac_key=mac_key,
+    )
+
+    envelope = ae_cipher.encrypt(plain)
+
+    encrypted = msgpack.packb({
+        'version': 1,
+        'salt': b'salt'*4,
+        'argon2_time_cost': 3,
+        'argon2_memory_cost': 2**16,
+        'argon2_parallelism': 4,
+        'argon2_type': b'id',
+        'algorithm': 'argon2 aes256-ctr hmac-sha256',
+        'data': envelope,
+    })
+    monkeypatch.setenv('BORG_PASSPHRASE', "hello, pass phrase")
+    passphrase = Passphrase.new()
+    key = KeyfileKey(None)
+
+    decrypted = key.decrypt_key_file(encrypted, passphrase)
+
+    assert decrypted == plain
+
+
+def test_decrypt_key_file_pbkdf2_sha256_aes256_ctr_hmac_sha256(monkeypatch):
+    monkeypatch.delenv('BORG_TESTONLY_MOCK_KDF')
```

Comment:
if we just tweaked the algo **params** for the tests like we did for pbkdf2 in the past, this line would not be needed and we still would test the real code of the algo (just not in its production strength).